### DISCUSSION TITLE
How to input a value to a joint at every step

### DISCUSSION THREAD

**Discussion by Creasento** [raisimTech/raisimLib#175] [Q&A] [answered]
I want to make feedback control to my manipulator.

So I found how to get joint variables (position, velocity, angular velocity) and save it.
Next step is how to input some variables (force in most cases).

Q1: If I want to make feedback control (such as DoB), can I write the code inside the "for" statement?
Because I want to control joints every step, so I think write code in for statement

for statement is this. (I append code)

```
for (int i = 0; i < runTime; i++) {

      std::this_thread::sleep_for(std::chrono::microseconds(1));
      //inside here?
      server.integrateWorldThreadSafe();
  }
```

Q2: If I want set torque to reroute joint, how can I? The document said only "setsetExternalTorque" API.
Then should I write "setGeneralizedForce"?
How to I set torque to joint?

**Comment by jhwangbo** [comment, marked as answer]
Q1. That is correct. Currently the contacts you get before ``integrateWorldThreadSafe()`` is previous contacts. I'll also push ``integrate1WorldThreadSafe()`` method so that you can get contacts of the current time step.
Q2. Yes. ``setGeneralizedForce`` is the way to go.

**Reply by Creasento** [reply]
If I set force like that, what is the unit of 1? N? or kg*m/s^2?
```
Eigen::VectorXd jointNominalConfig(arm1->getGeneralizedCoordinateDim());
  jointNominalConfig << 1, 0, 0, 0, 0;
  arm1->setGeneralizedForce(jointNominalConfig);
```

**Reply by jhwangbo** [reply]
https://raisim.com/sections/ArticulatedSystem.html#joints
